compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4374,6 +4377,64 @@ private STNode parseObjectTypeDescriptor() {
                 objectMembers, closeBrace);
     }
 
+    /**
+     * Parse object constructor expression.
+     *
+     * object-constructor-expr :=
+     *    [annots] [client] object [type-reference] {
+     *       object-member*
+     *    }
+     * object-member := object-field | method-defn
+     *
+     * @param isRhsExpr
+     * @param annots
+     * @return Parsed node
+     */
+    private STNode parseObjectConstructorExpressionNode(boolean isRhsExpr, STNode annots) {
+
+        startContext(ParserRuleContext.OBJECT_CONSTRUCTOR);
+
+        STNode objectTypeQualifier;
+        STToken nextToken = peek();
+        if (nextToken.kind == SyntaxKind.CLIENT_KEYWORD) {
+            startContext(ParserRuleContext.CLIENT_KEYWORD);
+            objectTypeQualifier = parseClientKeyword();
+            endContext();
+        } else {
+            objectTypeQualifier = STNodeFactory.createEmptyNode();
+        }
+
+        STNode objectKeyword = parseObjectKeyword();
+
+        STNode typeDescriptor;
+        nextToken = peek();
+        if (nextToken.kind == SyntaxKind.IDENTIFIER_TOKEN) {
+            typeDescriptor = parseTypeDescriptor(ParserRuleContext.TYPE_DESC_IN_NEW_EXPR);
+        } else {
+            typeDescriptor = STNodeFactory.createEmptyNode();
+        }
+
+        STNode objectCtorBody = parseObjectConstructorBody();
+
+        endContext();
+
+        return STNodeFactory.createObjectConstructorExpressionNode(annots,
+                objectTypeQualifier, objectKeyword, typeDescriptor, objectCtorBody);
+    }
+
+    private STNode parseObjectConstructorBody() {
+        startContext(ParserRuleContext.OBJECT_CONSTRUCTOR_BODY);
+
+        STNode openBrace = parseOpenBrace();
+        STNode objectMembers = parseObjectMembers();
```

Comment:
I think currently `parseObjectMembers()` allows `object-type-inclusion := * type-reference ;` as well. 
I can fix this when adding error recovery though. :)

Reply:
Do we have to duplicate functions or can we set a class level boolean to handle this easily?
As an example.
eg:
```Java
skipInclusion = true;
STNode objectMembers = parseObjectMembers();
skipInclusion = false;
```
Then conditionally handle at `parseObjectMemberWithoutMeta`.

Reply:
Class level variables can be dangerous since the parser is recursive-descent. 
eg: an anon-object inside the object constructor - then the flag is set to `true` when the parser reaches the inner construct, where it should have been `false`

Passing a flag to the method as a parameter would be fine though.

Reply:
Done with commit : df2ec9db2462323f69b6554eaa7446e660a652a7